pkg/kubelet/dockertools/manager.go

```diff
@@ -2058,13 +2063,25 @@ func getPidMode(pod *api.Pod) string {
 func getIPCMode(pod *api.Pod) string {
 	ipcMode := ""
 	if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.HostIPC {
-		ipcMode = "host"
+		ipcMode = namespaceModeHost
 	}
 	return ipcMode
 }
 
-// GetNetNs returns the network namespace path for the given container
-func (dm *DockerManager) GetNetNs(containerID kubecontainer.ContainerID) (string, error) {
+// IsUsingHostNetwork returns true if the provided container ID is using host
+// networking.  This method is used by network plugins to determine if a pod
+// requires additional setup. Pods running with host networking do not.
+func (dm *DockerManager) IsUsingHostNetwork(containerID kubecontainer.ContainerID) (bool, error) {
```

Comment:
Yeah things are too nascent right now to comment but that sounds reasonable. 

We could of couse not invoke the plugin for host network pods by putting code in the kubelet, I felt that was more icky.  We could also pass the host-network flag via env vars, another thing i'm not a huge fan of. 


I think ideally, we would not have a reverse channel from network plugin to kubelet. We exec the network plugin sending a json manifest through stdin, so that's a pretty hard boundry for isolation. The way we handle docker networking is essentially setup the interfaces for the pause container, and then join everything using `net=container:pause`, so really we're just handling setup for the single pause container.